### PR TITLE
Cleanup of meshing of fractured domains

### PR DESCRIPTION
**Pull request by keileg**
Main content:
1. Created a class FractureNetwork2d, and renamed FracturedNetwork-3d
2. All meshing of fractured domains run through these two functions - the old pp.meshing.simplex_grid is deleted.
3. Import filters for fracture networks are simplified
4. Old functionality for non-conforming dfn meshing is removed.

### CODE REVIEW

**Comment by keileg** [comment]
@alessiofumagalli : We should discuss whether the current implementation is compatible with non-convex domains. I think a small modification is needed, but I am not sure.

File: src/porepy/__init__.py [hunk 1]
```diff
@@ -72,8 +72,8 @@
 from porepy.grids.mortar_grid import MortarGrid, BoundaryMortar
 
 # Fractures
-from porepy.fracs.fractures import Fracture, EllipticFracture, FractureNetwork
-from porepy.fracs.meshing import simplex_grid
+from porepy.fracs.fractures import Fracture, EllipticFracture, FractureNetwork3d
```
Comment (alessiofumagalli) [inline src/porepy/__init__.py]: why FractureNetwork2d is in one module and FractureNetwork3d is in the standard fractures module?

File: test/unit/test_tag_gmsh.py [hunk 2]
```diff
@@ -84,17 +83,15 @@ def test_auxiliary(self):
 
     def test_auxiliary_refined(self):
         domain = {"xmin": 0, "xmax": 1, "ymin": 0, "ymax": 1}
-        fracs = []
+        network = pp.FractureNetwork2d(domain=domain)
```
[the remark this answers is not in the document]
Comment (keileg) [reply test/unit/test_tag_gmsh.py]: Good point. I will address this.

File: test/unit/test_tag_gmsh.py [hunk 3]
```diff
@@ -317,17 +309,15 @@ def test_fracture_auxiliary(self):
...
 
         subdomain = {
             "points": np.array([[0, 0.75, 0.5, 0.5], [0.5, 0.5, 0.25, 0.75]]),
             "edges": np.array([0, 1, 2, 3]).reshape((2, -1), order="F"),
         }
 
-        gb = pp.fracs.meshing.simplex_grid(
-            fracs, domain=domain, subdomains=subdomain, **mesh_args
-        )
+        gb = network.mesh(mesh_args, constraints=subdomain)
```
[the remark this answers is not in the document]
Comment (keileg) [reply test/unit/test_tag_gmsh.py]: I agree, subdomains is a left-over from the old days.

File: test/unit/test_fracture_intersect_boundary.py [hunk 4]
```diff
@@ -35,7 +36,7 @@ def test_completely_outside_lower(self):
         self.setup()
         f = self.f_1
         f.p[0] -= 2
-        network = FractureNetwork([f])
+        network = pp.FractureNetwork3d([f])
```
[the remark this answers is not in the document]
Comment (rbe051) [reply test/unit/test_fracture_intersect_boundary.py]: You can always check if it is a list and i not add a list.

It should at least throw an error if it does not get a list (or FractrueNetwork). Now it happily runs until it at some point crash and throw a cryptic error message about getting the wrong dimensions

**Comment by rbe051** [comment]
Is there a reason the coverage drops 1.2% ?

File: test/unit/test_fracture_intersect_boundary.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (keileg) [reply test/unit/test_fracture_intersect_boundary.py]: I will fix it.

**Comment by keileg** [comment]
> Is there a reason the coverage drops 1.2% ?

I can fix some of this. Also, the new FractureNetwork2d class has some utility functions that are not covered.

File: test/unit/test_fracture_intersect_boundary.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (keileg) [reply test/unit/test_fracture_intersect_boundary.py]: Single fractures can now pass without the []

**Pull request merged**